src/Text.js

```diff
@@ -10,8 +10,10 @@ export function extractLinks(fragments) {
             return f.split(UrlRegex).map(a => {
                 if (a.startsWith("http")) {
                     try {
-                        let url = new URL(a);
-                        let ext = url.pathname.toLowerCase().match(FileExtensionRegex);
+                        const url = new URL(a);
+                        const vParam = url.searchParams.get('v')
+                        const isYoutube = (url.host === "www.youtube.com" || url.host === "youtube.com" ) && vParam
```

Comment:
I would make this a separate function to iterate over fragments and have another Regex to parse the youtube links, can see this being the case for other sources in the future too